### PR TITLE
Enhance the setupLoadBalancer.sh and add integration test

### PR DESCRIPTION
**Pull request by anpanigr**
1. Added missing header to samples/scripts/domain-lifecycle/clusterStatus.sh
2. Added following intuitive ( position independent)  command line options to setupLoadBalancer.sh 
3. Added integration tests for setupLoadBalancer.sh 

$ setupLoadBalancer.sh -h
  Usage:
    setupLoadBalancer.sh -c[d]  -t ingress-type  [-n namespace] [-v version]
    -c                   : create ingress controller [required]
    -d                   : delete ingress controller [required]
    -t <ingress type>    : ingress type traefik, voyager or nginx [required]
    -v <ingress version> : ingress release version
    -n <namespace>       : ingress namespace
    -m <kubernetes_cli>  : Kubernetes command line interface. Default is 'kubectl' if KUBERNETES_CLI env variable is not set. Otherwise default is the value of KUBERNETES_CLI env variable.
    -h                   : print help

Jenking Job (parallel) 
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5382/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5423/

### CODE REVIEW

File: kubernetes/samples/scripts/domain-lifecycle/clusterStatus.sh [hunk 1]
```diff
@@ -1,4 +1,6 @@
-#!/bin/bash
+# !/bin/sh
```
Comment (tbarnes-us) [inline kubernetes/samples/scripts/domain-lifecycle/clusterStatus.sh:1]: Has it been confirmed taht this script works with 'sh' instead of 'bash'?

Comment (anpanigr) [reply kubernetes/samples/scripts/domain-lifecycle/clusterStatus.sh:1]: yes, the script works with sh. All the scripts in this directory has the inclusion of "# !/bin/sh"

File: kubernetes/samples/charts/util/setupLoadBalancer.sh [hunk 2]
```diff
@@ -26,297 +22,388 @@ DefaultTraefikVersion=2.2.1
...
+EOF
+exit $1
+}
+
+actionchoosen=false
+
+while getopts "cdt:n:r:v:h" opt; do
+  case $opt in
+    c) action="create"
+       if [ $actionchoosen = "true" ]; then
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply kubernetes/samples/charts/util/setupLoadBalancer.sh:74]: actionchoosen -> action_chosen

File: kubernetes/samples/charts/util/setupLoadBalancer.sh [hunk 3]
```diff
@@ -26,297 +22,388 @@ DefaultTraefikVersion=2.2.1
...
+    ;;
+    n) namespace="${OPTARG}"
+    ;;
+    t) ingressType="${OPTARG}"
+    ;;
+    r) release="${OPTARG}"
+    ;;
+    m) kubernetesCli="${OPTARG}"
+    ;;
+    h) usage 1
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply kubernetes/samples/charts/util/setupLoadBalancer.sh:95]: done

File: kubernetes/samples/charts/util/setupLoadBalancer.sh [hunk 4]
```diff
@@ -26,297 +22,388 @@ DefaultTraefikVersion=2.2.1
...
+              [[ -z "${release}"   ]] && release="${DefaultNginxVersion}"
+              [[ -z "${namespace}" ]] && namespace="${ingressType}"
+              repository="ingress-nginx"
+              chart="nginx-release"
+              ;;
+    *)         printError "Unsupported ingress type [${ingressType}]. Suppoprted ingress type are [traefik, voyager or nginx] "
+               exit -1  ;;
+esac
+
+printInfo "Action [${action}] Type [${ingressType}] NameSpace [${namespace}] Release [${release}] Chart [$chart]"
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply kubernetes/samples/charts/util/setupLoadBalancer.sh:135]: printInfo "Action [${action}], Type [${ingressType}], NameSpace [${namespace}], Release [${release}], Chart [$chart]"

File: kubernetes/samples/charts/util/setupLoadBalancer.sh [hunk 5]
```diff
@@ -26,297 +22,388 @@ DefaultTraefikVersion=2.2.1
...
 if [[ "$HELM_VERSION" =~ "v2" ]]; then
-  echo "Detected unsupported Helm version [${HELM_VERSION}]"
+  printError "Detected unsupported Helm version [${HELM_VERSION}]"
   exit -1
 fi
 
 function createNameSpace() {
  ns=$1
- namespace=`kubectl get namespace ${ns} 2> /dev/null | grep ${ns} | awk '{print $1}'`
+ namespace=`${kubernetesCli} get namespace ${ns} 2> /dev/null | grep ${ns} | awk '{print $1}'`
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply kubernetes/samples/charts/util/setupLoadBalancer.sh:152]: Refer to line #137
--
 Validate Kubernetes CLI availability
# Try to execute kubernetes cli to see whether cli is available
if ! [ -x "$(command -v ${kubernetesCli})" ]; then
   printError "${kubernetesCli} is not installed"
  exit -1
fi
----

File: kubernetes/samples/charts/util/setupLoadBalancer.sh [hunk 6]
```diff
@@ -26,297 +22,388 @@ DefaultTraefikVersion=2.2.1
...
+ingressType=""
+namespace=""
+release=""
+repository=""
+chart=""
+
+# timestamp
+#   purpose:  echo timestamp in the form yyyy-mm-ddThh:mm:ss.nnnnnnZ
+#   example:  2018-10-01T14:00:00.000001Z
+function timestamp() {
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply kubernetes/samples/charts/util/setupLoadBalancer.sh:35]: This utility script is self-contained.

File: kubernetes/samples/charts/util/setupLoadBalancer.sh [hunk 7]
```diff
@@ -26,297 +22,388 @@ DefaultTraefikVersion=2.2.1
...
 }
 
+function waitForIngressPod() {
+  type=$1
+  ns=$2
+
+  printInfo "Wait until ${type} ingress controller pod is running."
+  ipod=$(${kubernetesCli} -o name get po -n ${ns})
+  if [[ "${ipod}" != *$chart* ]]; then
+   printError "Couldn't find the pod associated with ${type} helm deployment. List helm deployment status on namespace [${ns}]. "
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply kubernetes/samples/charts/util/setupLoadBalancer.sh:166]: add --> helm list -n ${ns}

File: kubernetes/samples/charts/util/setupLoadBalancer.sh [hunk 8]
```diff
@@ -26,297 +22,388 @@ DefaultTraefikVersion=2.2.1
...
     helm repo add traefik https://containous.github.io/traefik-helm-chart
     helm repo update
   else
-    echo "Traefik chart repository is already added."
+    printInfo "Traefik chart repository is already added."
   fi
 
-  if [ "$(helm list --namespace $TSPACE | grep $TNAME |  wc -l)" = 0 ]; then
-    echo "Installing Traefik operator."
+  if [ "$(helm list -q -n ${ns} | grep $chart | wc -l)" = 0 ]; then
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply kubernetes/samples/charts/util/setupLoadBalancer.sh:264]: added 

set -eu
set -o pipefail

File: kubernetes/samples/charts/util/setupLoadBalancer.sh [hunk 9]
```diff
@@ -26,297 +25,402 @@ DefaultTraefikVersion=2.2.1
...
+    ;;
+    t) ingressType="${OPTARG}"
+    ;;
+    r) release="${OPTARG}"
+    ;;
+    m) kubernetesCli="${OPTARG}"
+    ;;
+    h) usage 0
+    ;;
+    * ) usage 0
```
Comment (ankedia) [inline kubernetes/samples/charts/util/setupLoadBalancer.sh:100]: The exit code should be 1 (i.e. `usage 1`) for unexpected argument.

**Review by sankarpn — APPROVED** [review]
LGTM

**Review by ankedia — APPROVED** [review]
LGTM.

**Pull request merged**